source/multisource.go

```diff
@@ -39,7 +39,7 @@ func (ms *multiSource) Endpoints(ctx context.Context) ([]*endpoint.Endpoint, err
 		}
 		if len(ms.defaultTargets) > 0 {
 			for i := range endpoints {
-				eps := endpointsForHostname(endpoints[i].DNSName, ms.defaultTargets, endpoints[i].RecordTTL, endpoints[i].ProviderSpecific, endpoints[i].SetIdentifier)
+				eps := endpointsForHostname(endpoints[i].DNSName, ms.defaultTargets, endpoints[i].RecordTTL, endpoints[i].ProviderSpecific, endpoints[i].SetIdentifier, "")
 				for _, ep := range eps {
 					ep.Labels = endpoints[i].Labels
 				}
```

Comment:
With two sources, the second one may erase labels of the first, no ? :thinking:

Reply:
How would that happen? The code, both before and after, is copying all the labels for each endpoint it is copying  (and applying `defaultTargets`. Each sub-source's endpoints will get their respective resource label.